WordPress/gutenberg
# Fix Image block lightbox missing alt attribute and improve accessibility

Addresses part of the issues from https://github.com/WordPress/gutenberg/issues/54971

## What?

- Fixes retrieval of the image alt attribute.
- Move the button to open the Lightboz _after_ the image.
- Better labelling for the Lightbox modal dialog: it must describe the dialog purpose, not the image itself.
- Removes `aria-hidden` from the closed modal dialog. This appears to be redundant as the closed modal dialog is already hidden from assistiv technology by the means of the `visibility: hidden` CSS property.
- Sets the `aria-modal` attribute dynamically instead of hardcoding on hte PHP side. This is also consistent with similar code in the navigation block.
- Sets the `aria-label` attribute dynamically.
- Increases the target size of the modal dialog Close button to be at least 40 by 40 pixels.
- Avoids to render an empty `role=""` attribute.
- Avoids to render an empty `src=""` attribute. A small base64 encoded gif is now used as 'placeholder' for the enlarged image.
- Minor docblocks improvements.

## Why?

- The retrieval of the image alt attribute to be added to the button to open the modal dialog was broken. Adding context to the button helps screen reader users understand what the button is about.
- Gaving the button before the image was not ideal for users who perceive content in a linearized way.
- The modal dialog labelling was set to the image alt text (if any) or the string 'Image' otherwise. This wasn not ideal, as the modal dialog labelling must explaiin what the dialog is about and not describe the image.
- Rendering incorrect or unnecessary `aria=*` attributes and `role` whren the modal dialog is closed is either invalid HTML or not ideal.
- The target size od the close button was really too small and dependent on inherited styles. 

## How?

See description above.

## Testing Instructions





- Create a post and add two images: one with alt attribute and the other one without.
- Set both images to open in the Ligthbox.
- Publish and see the front end.
- Inspect the source and make sure the button to open the modal dialog is labelled:
  - `Enlarge image` for the image without alt attribute.
  - `Enlarge image: {image alt text here}` for the image with alt attribute.
- Observe the button to open the modal dialog is now after the image in the DOM order.
- Open the modal dialog and observe the dialog is labelled `Enlarged image`.
- Observe the X close button size is not at least 40 by 40 pixels.

Additionally:
- After first page load and when the modal dialog is still closed, observe the modal dialog markup:
  - Does not have a `role` attribute.
  - Does not have an `aria-modal` attribute.
  - Does not have an `aria-hidden` attribute.
  - Does not have an `aria-label` attribute.
  - The `src` attribute of the enlarged image is a base64 encoded image.
- Open the modal dialog and observe that the dialog markup:
  - Does have a `role="dialog"` attribute.
  - Does have an `aria-modal="true"` attribute.
  - Does not have an `aria-hidden` attribute.
  - Does have an `aria-label="Enlarged image"` attribute.
  - The `src` attribute of the enlarged image is now the one of the actual enlarged image.


### Testing Instructions for Keyboard


## Screenshots or screencast
